pkg/arvo/app/btc-node-store.hoon

```diff
@@ -0,0 +1,149 @@
+::  btc-node-store: data store for state received from a bitcoin full node
+::
+::    data:            scry command:
+::
+::    default-wallet  .^(@t %gx /=btc-node-store=/default-wallet/noun)
+::    n-wallets       .^(@ud %gx /=btc-node-store=/n-wallets/noun)
+::
+::
+/-  *btc-node-store
+/+  *btc-node-json, default-agent, verb
+::
+=>  |%
+    ::
+    +$  card  card:agent:gall
+    ::
+    +$  state
+      $%  [%0 state-zero]
+      ==
+    ::
+    +$  state-zero
+      $:  =wallets
+          default-wallet=@t
+      ==
+    --
+::
+=|  state-zero
+=*  state  -
+::  Main
+::
+%+  verb  |
+^-  agent:gall
+=<  |_  =bowl:gall
+    +*  this      .
+        btc-core  +>
+        bc        ~(. btc-core bowl)
+        def       ~(. (default-agent this %|) bowl)
+    ::
+    ++  on-init
+      ^-  (quip card _this)
+      :-  ~
+      %_    this
+        wallets  (~(put by wallets) [*@t *wallet])
+      ==
+    ::
+    ++  on-save   !>(state)
+    ++  on-load
+      |=  old=vase
+      `this(state !<(state-zero old))
+    ::
+    ++  on-poke
+      |=  [=mark =vase]
+      ^-  (quip card _this)
+      ~&  got-poked-with-data=mark
+      |^
+      ?+    mark    (on-poke:def mark vase)
+          %btc-node-store-action
+        (store-action !<(btc-node-store-action vase))
+      ::
+          %btc-node-store-command
+        (store-command !<(btc-node-store-command vase))
+      ==
+      ::
+      ++  store-action
+        |=  action=btc-node-store-action
+        ^-  (quip card _this)
+        =^  cards  state
+          ?+  -.action  ~|([%unsupported-action -.action] !!)
+            %add-wallet     (handle-add:bc +.action)
+            %list-wallets   handle-list-wallet:bc
+            %update-wallet  (handle-update-wallet:bc +.action)
+          ==
+        [cards this]
+      ::
+      ++  store-command
+        |=  command=btc-node-store-command
+        ^-  (quip card _this)
+        =^  cards  state
+          ?+  -.command  ~|([%unsupported-command -.command] !!)
+              %switch-wallet  (handle-switch:bc +.command)
+          ==
+        [cards this]
+      --
+    ++  on-watch  on-watch:def
+    ++  on-leave  on-leave:def
+    ::  +on-peek: read from app state
+    ::
+    ++  on-peek
+      |=  =path
+      ^-  (unit (unit cage))
+      ?+  path  (on-peek:def path)
+          [%x %default-wallet ~]  ``noun+!>(default-wallet)
+          [%x %n-wallets ~]       ``noun+!>(~(wyt by wallets))
+      ==
```

Comment:
It was a way to control for when there are more than one wallets created, since then the endpoint format used to call the node needs to change. When we have only one wallet (on the BTC node) the calls that handle wallet logic have in this format: `http://127.0.0.1:8332/` but if there is more than one wallet we need to append `wallet`or it will fail: `http://127.0.0.1:8332/wallet/<THE_WALLET>`. I'm gonna check all of these again to make sure that this behavior holds.